File: code-examples/php/usage-examples/deleteOne.php
<?php
require_once __DIR__ . "/vendor/autoload.php";

// Replace the uri string with your MongoDB deployment's connection string
$uri = "mongodb://localhost:27017";

$client = new MongoDB\Client($uri);
$collection = $client->sample_mflix->movies;

// Delete the first document in the "movies" collection that matches
// the specified query document 
$filter = ['title' => 'Annie Hall'];
$options = ['limit' => 1];

try {
    $deleteResult = $collection->deleteOne($filter, $options);

    if ($deleteResult->getDeletedCount() === 1) {
        print("Successfully deleted one document.\n");
    } else {
        print("No documents matched the query. Deleted 0 documents.\n");
    }
} catch(MongoDB\Driver\Exception\Exception $e) {
    printf("Error: %s\n", $e->getMessage());
}

?>

File: code-examples/scala/usage-examples/deleteOne.scala
// Delete a document

import org.mongodb.scala._

// Replace the uri string with your MongoDB deployment's connection string
val uri: String = "<connection string uri>"

val mongoClient: MongoClient = MongoClient(uri)
val database: MongoDatabase = mongoClient.getDatabase("sample_mflix")
val collection: MongoCollection[Document] = database.getCollection("movies")

val query: Document = Document("title" -> "Annie Hall")

collection.deleteOne(query).subscribe(new Observer[DeleteResult] {
  override def onNext(result: DeleteResult): Unit = {
    if (result.getDeletedCount == 1) {
      println("Successfully deleted one document.")
    } else {
      println("No documents matched the query. Deleted 0 documents.")
    }
  }

  override def onError(e: Throwable): Unit = println(s"Error: $e")
  
  override def onComplete(): Unit = {
    println("Operation completed")
    mongoClient.close()
  }
})
